ui/packages/catalog-ui-search/src/main/webapp/behaviors/selection.behavior.js

```diff
@@ -0,0 +1,127 @@
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+/*global require*/
+const Behaviors = require('./Behaviors');
+const Marionette = require('marionette');
+
+function getMaxIndex(selectionInterface) {
+    const selectedResults = selectionInterface.getSelectedResults();
+    const completeResults = selectionInterface.getCompleteActiveSearchResults();
+    let maxIndex = -1;
+    selectedResults.forEach(function (result) {
+        maxIndex = Math.max(maxIndex, completeResults.indexOf(result));
+    });
+    return maxIndex;
+}
+
+function getMinIndex(selectionInterface) {
+    const selectedResults = selectionInterface.getSelectedResults();
+    const completeResults = selectionInterface.getCompleteActiveSearchResults();
+    if (selectedResults.length === 0) {
+        return -1;
+    }
+    return selectedResults.reduce(function (minIndex, result) {
+        return Math.min(minIndex, completeResults.indexOf(result));
+    }, completeResults.length);
+}
+
+const doubleClickTime = 500; // how soon a user has to click for it to be a double click
+const textSelectionTime = 200; // how long a user has to hold down mousebutton for us to recognize it as wanting to do text selection
+
+Behaviors.addBehavior('selection', Marionette.Behavior.extend({
+    events: function() {
+        return {
+            [`click ${this.options.selectionSelector}`]: 'handleClick',
+            [`mousedown ${this.options.selectionSelector}`]: 'handleMouseDown'
+        };   
+    },
+    lastTarget: undefined,
+    lastMouseDown: Date.now(),
+    lastClick: Date.now(),
+    onRender: function() {
+        this.view.$el.addClass('has-selection-behavior');
+    },
+    updateStateOnClick: function(event) {
+        this.lastTarget = event.currentTarget;
+        this.lastClick = Date.now();
+    },
+    handleClick: function(event) {
+        this.interpretClick(event);
+        this.updateStateOnClick(event);
+    },
+    interpretClick: function (event) {
+        if (event.altKey || this.isTextSelection() || this.isDoubleClick(event)) {
+            return;
+        }
+        const resultid = event.currentTarget.getAttribute('data-resultid');
+        const alreadySelected = this.options.selectionInterface.getSelectedResults().get(resultid) !== undefined;
+        //shift key wins over all else
+        if (event.shiftKey) {
+            this.handleShiftClick(resultid, alreadySelected);
+        } else if (event.ctrlKey || event.metaKey) {
```

Comment:
:question: Why is the use of a `Meta` key considered equivalent to a `Ctrl` here?

Reply:
For mac support.  On mac, by default it uses the `meta` key like the `ctrl` key.